Source/AwsCommonRuntimeKit/io/ClientBootstrap.swift

```diff
@@ -1,26 +1,26 @@
 //  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 //  SPDX-License-Identifier: Apache-2.0.
 import AwsCIo
-import Foundation
 
-public final class ClientBootstrap {
+public class ClientBootstrap {
     let rawValue: UnsafeMutablePointer<aws_client_bootstrap>
-    var enableBlockingShutdown: Bool = false
 
     public init(eventLoopGroup elg: EventLoopGroup,
                 hostResolver: HostResolver,
                 allocator: Allocator = defaultAllocator,
                 shutdownCallback: ShutdownCallback? = nil) throws {
         let shutdownCallbackCore = ShutdownCallbackCore(shutdownCallback)
         let shutdownOptions = shutdownCallbackCore.getRetainedShutdownOptions()
-        var options = aws_client_bootstrap_options(
-            event_loop_group: elg.rawValue,
-            host_resolver: hostResolver.rawValue,
-            host_resolution_config: hostResolver.config,
-            on_shutdown_complete: shutdownOptions.shutdown_callback_fn,
-            user_data: shutdownOptions.shutdown_callback_user_data
-        )
-        guard let rawValue = aws_client_bootstrap_new(allocator.rawValue, &options) else {
+        guard let rawValue = (withUnsafePointer(to: hostResolver.getHostResolutionConfig()) { hostResolutionConfigPointer in
+            var options = aws_client_bootstrap_options(
+                    event_loop_group: elg.rawValue,
+                    host_resolver: hostResolver.rawValue,
+                    host_resolution_config: hostResolutionConfigPointer,
+                    on_shutdown_complete: shutdownOptions.shutdown_callback_fn,
+                    user_data: shutdownOptions.shutdown_callback_user_data
+            )
+           return aws_client_bootstrap_new(allocator.rawValue, &options)
+        }) else {
```

Comment:
Nit: This can be a bit tricky to read, especially at first glance, given folks are used to seeing the contents of the `else` block right after the `guard` line. But here we have a closure definition, then the `else`.

Can we define a closure to encapsulate this logic to make the `guard` simpler to read?
Something like:

```suggestion
        let getRawValue: () -> UnsafeMutablePointer<aws_client_bootstrap> = {
            withUnsafePointer(to: hostResolver.getHostResolutionConfig()) { hostResolutionConfigPointer in
                var options = aws_client_bootstrap_options(
                        event_loop_group: elg.rawValue,
                        host_resolver: hostResolver.rawValue,
                        host_resolution_config: hostResolutionConfigPointer,
                        on_shutdown_complete: shutdownOptions.shutdown_callback_fn,
                        user_data: shutdownOptions.shutdown_callback_user_data
                )
               return aws_client_bootstrap_new(allocator.rawValue, &options)
        }
        
        guard let rawValue = getRawValue() else {
```

Reply:
Thanks, incorporated.